src/main/java/org/recommendation/data/MovieDataInitializer.java

```diff
@@ -0,0 +1,97 @@
+package org.recommendation.data;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.recommendation.data.helper.MovieDataHelper;
+import org.recommendation.log.Logger;
+import org.recommendation.log.SoutLogger;
+import org.recommendation.model.Genre;
+import org.recommendation.model.Movie;
+
+import java.io.BufferedReader;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.*;
+
+public class MovieDataInitializer extends DataInitializer{
+
+    private final Logger logger = new SoutLogger();
+    private Map<String, Movie> movieMap = new HashMap<>();
+    @Override
+    public Map<String, String[]> readAndCleanData(final BufferedReader br) {
+        Map<String,String[]> dataStore = new TreeMap<>();
+        String line;
+        try {
+            while ((line = br.readLine()) != null){
+                String[] data = line.split("\\|");
+                String [] filteredData = new String[data.length];
+                int counter=0;
+                for(int i=0;i<data.length;i++){
+                    if(data[i].equals("")){
+                        continue;
+                    } else if(i==1){
+                        int index = data[i].indexOf('(');
+                        String title = index!=-1 ?data[i].substring(0,index):data[i];
+                        String year = index!=-1? data[i].substring(index+1,index+5):data[i];
+                        filteredData[counter++]=title;
+                        filteredData[counter++]=year;
+                    }else{
+                        filteredData[counter++]=data[i];
+                    }
+                }
+                int totalItems = filteredData.length;
+                if(totalItems>2){
+                    dataStore.put(filteredData[0], Arrays.copyOfRange(filteredData, 1, totalItems));
+                    populateMovieData(filteredData);
+                }
+            }
+        }catch (Exception exception){
+            logger.error("MovieDataInitializer readAndCleanData()",exception);
+        }
+        MovieDataHelper.movieMap = Collections.unmodifiableMap(movieMap);
+        return dataStore;
+    }
+    private void populateMovieData(final String [] filteredData){
+        if(filteredData.length<5){
+            logger.info("Insufficient data to map, skipping this row "+ filteredData[0]);
```

Comment:
we can remove this check and put this method into try catch block so that if we all new column, we can easily add required handling without changes this if condition